Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,234 @@
+
+                for (int i = 1; i < userNameAndBalanceHistory.Length; i++)
+                {
+                    float balance;
+                    try
+                    {
+                        balance = float.Parse(userNameAndBalanceHistory[i]);
+                    }
+                    catch
+                    {
+                        balance = 0;
+                    }
+                    if (balance > highestBalance)
+                    {
+                        highestBalanceHolder = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+                    if (balance == highestBalance)
+                    {
+                        if (holderCount == 2)
+                        {
+                            highestBalanceHolder += ($" and {currentBalanceHolder}");
+                        }
+                        if (holderCount == 1)
+                        {
+                            highestBalanceHolder += ($", {currentBalanceHolder}");
+                            holderCount++;
+                        }
+                    }
+                    highestBalance = highestBalance < balance ? balance : highestBalance;
+                }
+            }
+            string stringHighestBalance = highestBalance.ToString();
+            return ($"{highestBalanceHolder} had the most money ever. ¤{stringHighestBalance}.");
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
-        }
+            float biggestLoss = 0f;
+            float balance;
+            float nextBalance;
+            string biggestLoser = "";
+            int holderCount = 0;
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userHistory = entry.Split(',');
+                if (userHistory.Length <= 2)
+                {
+                    return "N/A.";
+                }
+                string currentBalanceHolder = userHistory[0];
+
+                for (int i = 1; i < userHistory.Length - 1; i++)
+                {
+                    balance = float.TryParse(userHistory[i], out float result) ?
+                        float.Parse(userHistory[i]) : float.NaN;
+
+                    nextBalance = float.TryParse(userHistory[i + 1], out float nextResult) ?
+                        float.Parse(userHistory[i + 1]) : float.NaN;
 
+                    if ((nextBalance - balance) < biggestLoss)
+                    {
+                        biggestLoser = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+                    if (biggestLoss == nextBalance - balance)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)